classes/submission/reviewAssignment/ReviewAssignmentDAO.php

```diff
@@ -669,6 +669,41 @@ public function deleteByReviewRoundId($reviewRoundId): int
             ->map(fn ($reviewId) => $this->deleteById($reviewId))
             ->count();
     }
+
+    /**
+     * Delete all review assignments and review rounds for a given context ID.
+     *
+     * @param int $contextId
+     * @return void
+     */
+    public function deleteByContextId($contextId)
+    {
+        // retrieve all submission IDs for the context
+        $submissionIds = DB::table('submissions')
```

Comment:
There are two ways of going about this:

1. By getting a list of submission IDs, then iterating through them and deleting content one by one. (This is the approach you've taken.)
    - Rather than getting the list of submission IDs directly here, it would be better to rely on the submission code to do this:
        ```php
        $submissionIds = Repo::submission()->getCollector()->filterByContextIds([$contextId])->getIds();
        ```
    - Likewise, as I've mentioned below, it's better to delegate the review round deletion to the DAO responsible for that entity.
2. This would run faster, as it doesn't require iteration, but is riskier: code the deletion without iteration.
    ```php
    DB::table('review_assignments AS ra')->join('submissions AS s', 'ra.submission_id', '=', 's.submission_id')->where('s.context_id', $contextId)->delete();
    ```

I'm not 100% confident which approach is best in this case. The first approach will be slow, because a context might have a lot of submissions. But it's a rarely used operation, so performance isn't important. Optimizing the deletions to avoid iteration will be quick, but will not behave well with things like hook callbacks etc.